CMakeLists.txt

```diff
@@ -434,7 +434,9 @@ elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "riscv")
   endif()
 elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "(i.86|AMD64|x86_64|amd64)")
   set(X86 1)
-  if(HOST_DEVICE_ADDRESS_BITS MATCHES "32")
+  if(CMAKE_LIBRARY_ARCHITECTURE STREQUAL "x86_64-linux-gnux32")
+    set(X32 1)
```

Comment:
What is the effect of setting this variable? I'm not seeing or finding anything that could act on this variable.

Reply:
`X32` itself is not used anywhere (and `X86_64` does not seem to be used anywhere either), but the effect is `I386` no longer getting set in this case, and therefore some i386 specific code paths no longer being used on x32

Reply:
this change is sufficient to make PoCL  build on x32, but there are still some failing tests, but that is something I want to investigate later (after switching the Debian package to 6.1 and llvm-19)